src/mococrw/key.h

```diff
@@ -91,6 +92,12 @@ class AsymmetricPublicKey
      */
     static AsymmetricPublicKey readPublicKeyFromPEM(const std::string& pem);
 
+    /**
+     * Loads a public key from an HSM, creating an @ref AsymmetricPublicKey
+     * object as a result.
+     */
+    static AsymmetricPublicKey readPublicKeyFromHSM(HSM &hsm, const std::string& keyID);
```

Comment:
I'm not sure if this is the best since as this function internally calls `ENGINE_load_public_key(ENGINE *e, const char *key_id, UI_METHOD *ui_method, void *callback_data)`.

Reply:
Why should that matter? I don't see `hsm` object being changed anywhere in the call graph to warrant non-consting it.

Reply:
Because an HSM is an abstraction of `Engine *e` and the `ENGINE_load_public_key` prototype does not guarantee that the engine referred to by `e` is not modified by said function.

Reply:
In `hsm` object you are holding a pointer to the `ENGINE` object and that pointer shouldn't change. Or should it?

Reply:
I'm not sure why users of this API would care what happens with underlying `ENGINE` object that is referenced by `hsm` object. Will user at any point want to know what's going on in the underlying `ENGINE` object? This is what you're trying to abstract. AFAIC we can resolve this conversation since there are probably more important things that have to be discussed first and there are other instances where const is not used correctly.